pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -680,37 +682,36 @@ func (r *shootReconciler) patchShootStatusOperationSuccess(
 		now                        = metav1.NewTime(time.Now().UTC())
 		description                string
 		setConditionsToProgressing bool
-		err                        error
-		isHibernated               = v1beta1helper.HibernationIsEnabled(shoot)
 	)
 
 	switch operationType {
 	case gardencorev1beta1.LastOperationTypeCreate, gardencorev1beta1.LastOperationTypeReconcile:
-		description = "Shoot cluster state has been successfully reconciled."
+		description = "Shoot cluster has been successfully reconciled."
 		setConditionsToProgressing = true
 
 	case gardencorev1beta1.LastOperationTypeMigrate:
-		description = "Shoot cluster state has been successfully migrated."
-		setConditionsToProgressing = true
+		description = "Shoot cluster has been successfully prepared for migration."
+		setConditionsToProgressing = false
```

Comment:
Why was this set to `false`?

Reply:
Simply to preserve the previous behavior:
* This method was never previously invoked with `operationType` set to `LastOperationTypeMigrate` (so the above 2 lines were basically dead code).
* The code in `finalizeShootPrepareForMigration` that was deleted and replaced with an invocation of this method didn't set any conditions.

Do you think it should be set to `true` and therefore change the current behavior? Honestly I am not really sure why this is being done for `Reconcile` and `Restore` in the first place - don't we rely on the "shoot care" controller to set these conditions properly? Why do we revert any of them from e.g. `True` to `Progressing` on successful reconciliation / restoration?

Reply:
OK, thanks for the explanation, I'm fine for now if this was the previous behaviour (just got curious since it wasn't obvious to me). 

> why this is being done for Reconcile and Restore in the first place - don't we rely on the "shoot care" controller to set these conditions properly? Why do we revert any of them from e.g. True to Progressing on successful reconciliation / restoration?

This was introduced with https://github.com/gardener/gardener/pull/3251 (Issue: https://github.com/gardener/gardener/issues/3238):

> The shoot reconciler sets the conditions to Progressing after it finished a successful reconciliation, and the care controller starts to re-evaluate the health status after this happened. This helps end-users to better understand whether their cluster is indeed healthy after a reconciliation. Earlier, it could take up to 30s/1m (based on the configured care controller sync period) until the actual status is reflected.

I'm not sure if that's relevant for the migration case as well, no strong opinion here.

Reply:
I am also not sure, so I would suggest to keep the old behavior for now.